### PR TITLE
fix the connectedCheck build target

### LINKED ISSUE #2871: Unable to run instrumentation tests due to proguard
Running `./gradlew proguardDebugTest` now fails (probably an issue with the [refactor](https://github.com/WhisperSystems/TextSecure/commit/597f7bbafd2ff7ef2779ca5292a2fc8ef568d489)).

The log is huge so I've put it on pastebin: http://pastebin.com/rkQ0BxUZ

### PR DESCRIPTION
**Pull request by rhodey**
1) create a new build type used exclusively for testing, 'testing'
2) only obfuscate the package android.support.v7.internal.view.menu to
prevent LGE ROM bug
3) '-keepattributes Exceptions' to allow for throwing from mocks
4) -dontskipnonpubliclibraryclassmembers and -dontwarn for everything else

Fixes #2871
// FREEBIE

### CODE REVIEW

**Comment by mcginty** [comment]
not tested, but code lgtm. surprisingly small diff given how gradle changes usually look.

File: proguard-appcompat-v7.pro [hunk 1]
```diff
@@ -1,7 +1,8 @@
+# only obfuscate this one pagacke because LGE ROM bug
+-keepnames class !android.support.v7.internal.view.menu.**, ** { *; }
+
 -keep public class android.support.v7.widget.** { *; }
 -keep public class android.support.v7.internal.widget.** { *; }
-# below line disabled due to LGE ROM bug.
-# -keep public class android.support.v7.internal.view.menu.** { *; }
```
Comment (moxie0) [inline proguard-appcompat-v7.pro]: Have you tested this change on LGE devices to ensure it didn't break anything?

**Comment by rhodey** [comment]
I am unable to reproduce the appcompat MenuBuilder crash bug documented internally at #2116 and externally [here with google](https://code.google.com/p/android/issues/detail?id=78377)

@mcginty you mentioned that you used [this uiautomator test](https://gist.github.com/mcginty/da0e49636109b428ff91) and an LG Optimus L9 from AppThwack to test your fix for this bug. However this issue exists only for Android 4.2.2 and the L9 is a 4.1.2 device-- also uiautomator tests can only be run on devices API level 18 and above so I can't understand how this happened. Are you positive that you tested using this device? It seems more likely that you used the LG 9 to test [android issue 78154](https://code.google.com/p/android/issues/detail?id=78154), though I still don't understand how you got around the API 18 limit.

Even though I had doubts I spent several hours testing against the LG Optimus L9 using Espresso and was unable to reproduce this bug after verifying that the `android.support.v7.internal.view.menu` classes were both included in the output APK and not obfuscated.

Developers in the google issue mention repeatedly that the Samsung Galaxy Light running 4.2.2 was an ideal test vector for this bug on AppThwack, so I tried that. Using an APK where all `android.support.v7.internal.view.menu` classes were both included in the output APK and not obfuscated this bug still did not occur.

Developers in that issue also claim that Wiko Rainbow devices on 4.2.2 also reliably experience this bug. TestObject happens to have one of these devices so I tried, same as before, to reproduce this bug both using Espresso automated tests and their manual VNC-style web interface. Neither of these tests reproduced the bug, everything ran normally. Note: this bug is known to occur in `com.android.support:appcompat-v7:21.0.3`, our appcompat version.

Our commit history seems to indicate that two things were done to resolve #2116. First, reverting appcompat to v20 in bf69a90b6937f8d3b370f84272261a383987e4cc, then later updating to appcompat v21 and adding the proguard obfuscation trick in 810abe0275364f11c9f1672a385048c82ba23b4b. Please let me know exactly how you tested 810abe0275364f11c9f1672a385048c82ba23b4b to reproduce the crash documented in #2116, I've tried everything that has been publicly documented.

**Comment by rhodey** [comment]
@mcginty was able to confirm that his tests had actually been run on Samsung Galaxy Light and that 810abe0275364f11c9f1672a385048c82ba23b4b was the last time he was able to reproduce this bug, so I checked out 810abe0275364f11c9f1672a385048c82ba23b4b and edited the proguard rule such that it should crash.

```
-keep class !android.support.v7.internal.view.nope.**,** {*;}
```

Then built the debug target and uploaded it to AppThwack for testing, where it failed as I hoped. Finally I modified the proguard rule to work the same as this pull and ran the test again on AppThwack.

```
-keepnames class !android.support.v7.internal.view.menu.**, ** { *; }
```

And it worked! So, my conclusions are that:

1) this bug no longer exists in our builds with our without proguard obfuscation
2) if this bug was to ever resurface in our builds, the proguard obfuscation provided by this pull would prevent it from surfacing
3) the proguard comment should be updated to reference Samsung ROMs, the LGE ROM bug is a whole different issue related to hardware menu buttons and is not solved or solvable by proguard

failing test >> https://appthwack.com/public/56fd960425f374d1dc1e5428b?view_mode=by_device
passing test >> https://appthwack.com/public/275b81885f037fa331d534946?view_mode=by_device

**Comment by moxie0** [comment]
in 2.13.0

**Pull request closed without merging**